src/devices/machine/mm5740.cpp

```diff
@@ -0,0 +1,566 @@
+// license:BSD-3-Clause
+// copyright-holders:R. Belmont, Mark Garlanger
+/**********************************************************************
+
+    Nathional Semiconductor MM5740 Keyboard Encoder emulation
+    (Code based on kb3600.cpp)
+
+*********************************************************************/
+
+#include "mm5740.h"
+
+
+//**************************************************************************
+//  GLOBAL VARIABLES
+//**************************************************************************
+
+// devices
+const device_type MM5740 = &device_creator<mm5740_device>;
+
+//**************************************************************************
+//  LIVE DEVICE
+//**************************************************************************
+
+//-------------------------------------------------
+//  mm5740_device - constructor
+//-------------------------------------------------
+
+mm5740_device::mm5740_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock)
+	: device_t(mconfig, MM5740, "MM5740", tag, owner, clock, "mm5740", __FILE__),
+	m_read_x1(*this),
+	m_read_x2(*this),
+	m_read_x3(*this),
+	m_read_x4(*this),
+	m_read_x5(*this),
+	m_read_x6(*this),
+	m_read_x7(*this),
+	m_read_x8(*this),
+	m_read_x9(*this),
+	m_read_shift(*this),
+	m_read_control(*this),
+	m_write_data_ready(*this)
+{
+	for (auto & elem : m_x_mask)
```

Comment:
I know legacy code should be cleaned up.  That doesn't mean new code shouldn't be written properly.